### ISSUE TITLE
metrics for rebase feature

### ISSUE THREAD

**Issue by shiftkey** [desktop/desktop#6550]
This is a placeholder issue for discussing what metrics we want to roll out as part of this feature. There's probably some prior art from the Merge Conflicts feature that we can mirror here.

cc @desktop/product @desktop/analytics

**Comment by shiftkey** [comment]
@desktop/product @desktop/analytics friendly nudge on what metrics should be in for 1.6.3, versus what can be deferred to 1.7.0

**Comment by scottypate** [comment]
Deferring that decision to @iAmWillShepherd. I would lean on the side of making sure what we have is accurate prior to focusing on new metric additions.

**Comment by billygriffin** [comment]
@scottypate I view that project as more of a thing that's happening in parallel but shouldn't stop us from continuing to measure the things we're shipping.

@shiftkey For 1.6.3, I think we ought to measure:

1. Number of times a user pulls with rebase (potentially as a % of regular pulls if that's possible?)
2. Number of times a user encounters a conflict while pull rebasing.
2a. Number of times a pull rebase is completed successfully after a conflict.
2b. Number of times a pull rebase is aborted after a conflict.
3. Number of times a user force pushes (potentially as a % of regular pushes if that's possible?)

The big things we're trying to learn here are whether this is something people are enabling without surfacing anything in the UI, how frequently they're hitting conflicts, and whether they're subsequently force pushing the resolution.

Does that sound reasonable/appropriate?

**Comment by shiftkey** [comment]
> Does that sound reasonable/appropriate?

This all sounds reasonable

**Comment by scottypate** [comment]
`Does that sound reasonable/appropriate?`

I second that this is all good stuff. 

A follow-up measurement that I need to look at is the effect that rebasing in desktop has on other collaborators on the repo that are not using desktop. I have a hypothesis that easier rebasing in Desktop could increase conflicts in other areas of the product. We don't need to collect any data for that here other than what @billygriffin described above.

**Comment by shiftkey** [comment]
> 3\. Number of times a user force pushes (potentially as a % of regular pushes if that's possible?)

Force push is not in scope for `1.6.3` so we don't have anything to support this metric. Moving this out to #6935 to track there.

**Comment by shiftkey** [comment]
An update on what is coming in `1.6.3` for rebase metrics:

 -  Number of times a user pulls with rebase (potentially as a % of regular pulls if that's possible?)

See #6893 for those metrics

https://github.com/desktop/desktop/blob/47afaff0bd2cd389ead699866ddf5bcef2cfc0d6/app/src/lib/stores/app-store.ts#L2868-L2872

2. Number of times a user encounters a conflict while pull rebasing.
   2a. Number of times a pull rebase is completed successfully after a conflict.
   2b. Number of times a pull rebase is aborted after a conflict.

See #6951 for those metrics

https://github.com/desktop/desktop/blob/5a6d6024caca5551bcac4003f6f055fa6ae5f590/app/src/lib/stores/updates/changes-state.ts#L218-L225

Once both of those are merged, I'll add docs and update Central to pass through these new metrics so we can get them into the beta.
